echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/config/SlackConfig.groovy

```diff
@@ -44,10 +44,24 @@ class SlackConfig {
   final static String SLACK_INCOMING_WEBHOOK = 'https://hooks.slack.com'
   final static String SLACK_CHAT_API = 'https://slack.com'
 
+  @Value('${slack.base-url:}')
+  String slackBaseUrl
+
+  @Value('${slack.compatibleAPI:false}')
+  Boolean slackCompatibleAPI
```

Comment:
I am struggling to understand what is the meaning for `slackCompatibleAPI`?
I think it would be more appropriately named `forceUseIncomingWebhook`.
Also, probably worth having some exception to make sure end users understand that they either specify `slackBaseUrl` or `slackCompatibleAPI/forceUseIncomingWebhook` not both

Reply:
but if it makes sense to keep it as is, out convention is:
```
@Value('${slack.compatible-api:false}')
```

Reply:
Can you recommend a way to do that type of config validations? I searched the codebase but couldn't find any example.

Reply:
slackCompatibleAPI comes from the previous comment from @emjburns. I'll change it to forceUseIncomingWebhook as it makes more sense in the context.
In our use case, we'd need to actually specify both, since we need the webhook behaviour (Which defaults to false) and the base url.

Reply:
I think Spring provides some validation mechanism, but it might be easier to just `throw` inside the `slackEndoint`/`slackService` bean functions

Reply:
I've changed the name of the config value and also extracted the token detection to its own method, and I think it's more clear now. I wasn't really convinced with the previous name. I think it's now clear that either you force the setting or it will be detected from the token.
The change of the base url then becomes something independent of the incoming webhook setting.